File: CarePackage/Software/MinecraftLauncher.cs
namespace CarePackage.Software;

public class MinecraftLauncher : BaseSoftware
{
    public override string           Key            { get; set; } = "minecraft-launcher";
    public override string           Name           { get; set; } = "Minecraft Launcher";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Gaming;
    public override string           DownloadName   { get; set; } = "MinecraftInstaller.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.minecraft_launcher;
    public override string           Homepage       { get; set; } = "https://minecraft.net";
    
    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://launcher.mojang.com/download/MinecraftInstaller.exe?ref=mcnet");
}

File: packages/app/src/services/software/definitions/variants/minecraftLauncher.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'minecraft-launcher-latest';
	public name = 'Minecraft Launcher';
	public category = [SoftwareCategory.Gaming];
	public downloadName = 'MinecraftInstaller.exe';
	public icon = 'minecraft-launcher.png';
	public homepage = 'https://minecraft.net';
	public parent = 'minecraft-launcher';

	public async resolveDownloadUrl() {
		return ok('https://launcher.mojang.com/download/MinecraftInstaller.exe?ref=mcnet');
	}
}
